sdk/containerregistry/azure-containers-containerregistry/README.md

```diff
@@ -89,32 +116,110 @@ ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+PagedIterable<ArtifactTagProperties> tags = image.listTags();
+
+System.out.printf(String.format("%s has the following aliases:", image.getFullyQualifiedName()));
+
+for (ArtifactTagProperties tag : tags) {
+    System.out.printf(String.format("%s/%s:%s", anonymousClient.getEndpoint(), repositoryName, tag.getName()));
+}
+```
 
-client.listRepositoryNames().subscribe(repository -> System.out.println(repository));
+### Set artifact properties
+
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L116-L129 -->
+```Java
+TokenCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+
+ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(defaultCredential)
+    .buildClient();
+
+RegistryArtifact image = client.getArtifact(repositoryName, digest);
+
+image.updateTagProperties(
+    tag,
+    new ArtifactTagProperties()
+        .setWriteEnabled(false)
+        .setDeleteEnabled(false));
+```
+
+### Delete Images
+
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L82-L110 -->
+```Java
+TokenCredential defaultCredential = new DefaultAzureCredentialBuilder().build();
+
+ContainerRegistryClient client = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .credential(defaultCredential)
+    .buildClient();
+
+final int imagesCountToKeep = 3;
+for (String repositoryName : client.listRepositoryNames()) {
+    final ContainerRepository repository = client.getRepository(repositoryName);
+
+    // Obtain the images ordered from newest to oldest
+    PagedIterable<ArtifactManifestProperties> imageManifests =
+        repository.listManifests(
+            ManifestOrderBy.LAST_UPDATED_ON_DESCENDING,
+            Context.NONE);
+
+    imageManifests.stream().skip(imagesCountToKeep)
+        .forEach(imageManifest -> {
+            System.out.printf(String.format("Deleting image with digest %s.%n", imageManifest.getDigest()));
+            System.out.printf("    This image has the following tags: ");
+
+            for (String tagName : imageManifest.getTags()) {
+                System.out.printf("        %s:%s", imageManifest.getRepositoryName(), tagName);
+            }
+
+            repository.getArtifact(imageManifest.getDigest()).delete();
+        });
+}
+```
+
+### Delete repository with anonymous access throws 
+<!-- embedme ./src/samples/java/com/azure/containers/containerregistry/ReadmeSamples.java#L152-L164 -->
+```Java
+final String endpoint = getEndpoint();
+final String repositoryName = getRepositoryName();
+
+ContainerRegistryClient anonymousClient = new ContainerRegistryClientBuilder()
+    .endpoint(endpoint)
+    .buildClient();
+
+try {
+    anonymousClient.deleteRepository(repositoryName);
+    System.out.println("Unexpected Success: Delete is not allowed on anonymous access");
+} catch (Exception ex) {
```

Comment:
Use `ClientAuthenticationException` here to show that it throws a specific exception that is diagnosable.